pulley/src/interp/tail_loop.rs

```diff
@@ -29,9 +29,14 @@
 use super::*;
 use crate::decode::{unwrap_uninhabited, ExtendedOpVisitor};
 use crate::opcode::Opcode;
+use crate::profile::ExecutingPcRef;
 use crate::ExtendedOpcode;
 
-type Handler = fn(Interpreter<'_>) -> Done;
+/// ABI signature of each opcode handler.
+///
+/// Note that this "explodes" the internals of `Interpreter` to individual
+/// arguments to help get them all into registers.
+type Handler = fn(&mut MachineState, UnsafeBytecodeStream, ExecutingPcRef<'_>) -> Done;
```

Comment:
I'll note that this change was done to ensure/guarantee that these three components of `Interpreter` are passed in registers. I was worried about crossing a threshold where if `Interpeter` got too big it would be passed by-ref instead of "exploded" into components like we want.

Reply:
I'll also note that `ExecutingPcRef` is a zero-sized-type when `profile` is disabled, otherwise it's a pointer-large.

Reply:
Makes sense, thanks for the explanation.